lms/djangoapps/courseware/user_state_client.py

```diff
@@ -193,7 +154,10 @@ def set_many(self, username, block_keys_to_state, scope=Scope.user_state):
         else:
             user = User.objects.get(username=username)
 
-        evt_time = time()
+        if user.is_anonymous():
+            # Anonymous users cannot be persisted to the database, so let's just use
+            # what we have.
+            return
```

Comment:
@ormsbee Our load testing determined that the current CSM implementation would handle the next two years of load if we doubled the load each of those years. So there's no immediate need to track these stats for load purposes. That's why I approved the removal. If you feel that continuing to capture the data would be useful, I'm fine with keeping the tracking.